src/Storages/NamedCollectionsHelpers.h

```diff
@@ -16,16 +16,82 @@ namespace ErrorCodes
 namespace DB
 {
 
-NamedCollectionPtr tryGetNamedCollectionWithOverrides(ASTs asts);
+/// Helper function to get named collection for table engine.
+/// Table engines have collection name as first argument of ast and other arguments are key-value overrides.
+MutableNamedCollectionPtr tryGetNamedCollectionWithOverrides(
+    ASTs asts, ContextPtr context, bool throw_unknown_collection = true, std::vector<std::pair<std::string, ASTPtr>> * complex_args = nullptr);
+/// Helper function to get named collection for dictionary source.
+/// Dictionaries have collection name as name argument of dict configuration and other arguments are overrides.
+MutableNamedCollectionPtr tryGetNamedCollectionWithOverrides(const Poco::Util::AbstractConfiguration & config, const std::string & config_prefix);
 
 HTTPHeaderEntries getHeadersFromNamedCollection(const NamedCollection & collection);
 
-template <typename RequiredKeys = std::unordered_set<std::string>,
-          typename OptionalKeys = std::unordered_set<std::string>>
+struct ExternalDatabaseEqualKeysSet
+{
+    static constexpr std::array<std::pair<std::string_view, std::string_view>, 5> equal_keys{
+        std::pair{"username", "user"}, std::pair{"database", "db"}, std::pair{"hostname", "host"}, std::pair{"addresses_expr", "host"}, std::pair{"addresses_expr", "hostname"}};
```

Comment:
you don't have to write `std::pair` inside array

Reply:
you mean making it as `{"username", "user"}` instead of `std::pair{"username", "user"}`?

Reply:
yes, does it work?